src/prefect/engine.py

```diff
@@ -213,6 +227,7 @@ async def create_then_begin_flow_run(
     engine_logger.info(f"Created flow run {flow_run.name!r} for flow {flow.name!r}")
 
     if state.is_failed():
+        flow_run_logger(flow_run).error(validation_failure_msg)
```

Comment:
I think it could be easy for this variable name to change, and so instead of relying on a variable set within conditional code paths above I think you'll find this to be more robust if you did this instead:
```suggestion
        flow_run_logger(flow_run).error(state.message)
```